### PR TITLE
feat: Improve kafka rate calculations

### PR DESCRIPTION
**Pull request by spmason**
..using statistics(tm)..

This changes our approach to calculating topic and consumer rates, so that we use the raw rate of each rather than keeping a list of offsets

We use the 99th percentile of the recorded values, and Consumer rates are normalized to a single consumer so we can estimate the rate with an arbitrary number of consumers

### CODE REVIEW

File: src/test/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/LagModelTest.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    private static ExpectedResults expect(int replicaCount, long lag, OptionalDouble consumerRate, OptionalDouble topicRate) {
+        return new ExpectedResults(replicaCount, lag, consumerRate, topicRate);
+    }
+//
+//    @ParameterizedTest
+//    @MethodSource
+//    public void estimateLoadedConsumerRate(int previousReplicaCount,
+//                                      List<Map<TopicPartition, Long>> previousConsumerOffsets,
```
Comment (StephenFlavin) [inline src/test/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/LagModelTest.java:138]: are these tests supposed to be commented out?

File: src/main/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/TopicConsumerStats.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        return OptionalDouble.of(rate);
+    }
+
+    public OptionalDouble estimateConsumerRate(int replicaCount) {
+        var rate = historicalConsumerRates.getPercentile(99D);
+
+        if (!Double.isFinite(rate)) {
+            return OptionalDouble.empty();
+        }
```
Comment (StephenFlavin) [inline src/main/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/TopicConsumerStats.java:70]: I've never seen this used before? what the issue if the rate isn't finite?

File: src/test/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/LagModelTest.java (hunk 1 above)
Comment (spmason) [reply src/test/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/LagModelTest.java:138]: Ah, meant to delete these 🤦

File: src/main/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/TopicConsumerStats.java (hunk 2 above)
Comment (spmason) [reply src/main/java/com/brandwatch/kafka_pod_autoscaler/triggers/kafka/TopicConsumerStats.java:70]: If you divide a Double by zero then you get `Infinity`, which then can't really be used for any calculations. SImilar with `NaN` (Not A Number). Best to treat these as missing stats

**Review by StephenFlavin — APPROVED** [review]

**Pull request merged**